j5/vision/orientation.py

```diff
@@ -0,0 +1,84 @@
+"""Orientation classes to represent rotations in space."""
+
+from typing import List, Tuple
+
+from pyquaternion import Quaternion
+
+
+class Orientation:
+    """
+    Represents an orientation in 3D space.
+
+    Uses a unit quaternion as an internal representation.
+    """
+
+    def __init__(self, orientation: Quaternion):
+        self._quaternion = orientation
```

Comment:
Can we expose the quaternion as a property? I can imagine some competitors may well find a use for it.

```python
@property
def quaternion(self):
    return self._quaternion
```